src/ch/imedias/rsccfx/model/Rscc.java

```diff
@@ -154,22 +165,60 @@ public void requestKeyFromServer() {
     keyServerSetup();
 
     String command = commandStringGenerator(
-        pathToResourceDocker, "start_x11vnc.sh", pathToStunDumpFile);
+        pathToResourceDocker, "port_share.sh", vncPort.getValue(), pathToStunDumpFile);
     String key = systemCommander.executeTerminalCommand(command);
     setKey(key); // update key in model
+    startVncServer();
   }
 
   /**
    * Starts connection to the user.
    */
   public void connectToUser() {
     keyServerSetup();
+    String command = commandStringGenerator(pathToResourceDocker,
+        "port_connect.sh", vncPort.getValue(), getKey());
+    systemCommander.executeTerminalCommand(command);
+    startVncViewer("localhost");
+  }
+
+  /**
+   * Starts the VNC Server.
+   */
+  public void startVncServer() {
+    StringBuilder vncServerAttributes = new StringBuilder("-bg -nopw -q -localhost");
+
+    if (vncOptionViewonly.getValue()) {
+      vncServerAttributes.append(" -viewonly");
+    }
+    if (vncOptionWindow.getValue()) {
+      vncServerAttributes.append(" -sid pick");
+    }
+    vncServerAttributes.append(" -rfbport " + vncPort.getValue());
 
-    String command = commandStringGenerator(pathToResourceDocker, "start_vncviewer.sh", getKey());
+    String command = commandStringGenerator(null,
+        "x11vnc", vncServerAttributes.toString());
     systemCommander.executeTerminalCommand(command);
   }
 
   /**
+   * Starts the VNC Viewer.
+   */
+  public void startVncViewer(String hostAddress) {
+    if (hostAddress == null) {
+      throw new IllegalArgumentException();
+    }
+    StringBuilder vncViewerAttributes = new StringBuilder("-encodings copyrect ")
+        .append(" ").append(hostAddress);
+    //Encodings are missing: "tight zrle hextile""
```

Comment:
vncviewer support more encoding formats, but the system commander does not work with the list. I make a todo from that , as i don't know yet what is needed.